src/common/dataHooks/useMutationWithNotification.js

```diff
@@ -33,6 +34,12 @@ export default function useMutationWithNotification({
           navigate(redirectTo)
         }, 1000)
       }
```

Comment:
To prevent this hook from becoming overly complicated, you could pass the conditional refresh action as a callback function via `onSuccessCallback` . Also, is there another way we could update the state of the app without refreshing?

As it isn't being implemented at this stage I would suggest just removing this `refresh` modification and we can see if there is an alternative to refreshing the page when we implement it.